EpikCord/client/websocket_client.py

```diff
@@ -265,7 +308,111 @@ def stop_loop_on_completion(f: asyncio.Future):
             pass
         finally:
             future.remove_done_callback(stop_loop_on_completion)
-            self.utils.cleanup_loop(loop)
+            Utils.cleanup_loop(loop)
+
+
+
+    async def _voice_server_update(self, data: discord_typings.VoiceServerUpdateEvent):
+        voice_data: discord_typings.VoiceServerUpdateEvent = data["d"]
+        payload = {
+            "token": voice_data["token"],
+            "endpoint": voice_data["endpoint"],
+            "guild_id": voice_data["guild_id"],
+        }
+
+        if voice_data["endpoint"]:
+            payload["endpoint"] = voice_data["endpoint"]
+
+        return await self.dispatch("voice_server_update", payload)
+
+    async def _voice_state_update(self, data: discord_typings.VoiceStateUpdateEvent):
+        from EpikCord import VoiceState
+
+        return await self.dispatch(
+            "voice_state_update", VoiceState(self, data)
+        )  # TODO: Make this return something like (VoiceState, Member) or make VoiceState get Member from member_id
+
+    async def _guild_members_chunk(self, data: discord_typings.GuildMembersChunkEvent):
+        ...
+
+    async def _guild_delete(self, data: discord_typings.GuildDeleteEvent):
+        guild = self.guilds.remove_from_cache(data["id"])  # type: ignore
+
+        if guild:
+            await self.dispatch("guild_delete", guild)
+
+    async def _interaction_create(self, data: discord_typings.InteractionCreateEvent):
+        interaction = Utils.interaction_from_type(data)
+        await self.dispatch("interaction_create", interaction)
+
+    async def _channel_create(self, data: discord_typings.ChannelCreateEvent):
+        channel = self.utils.channel_from_type(data)  # type: ignore
+        self.channels.add_to_cache(channel.id, channel)  # type: ignore
+        return channel
+
+    async def _message_create(self, data: discord_typings.MessageCreateEvent):
+        """Event fired when messages are created"""
+        from EpikCord import Message
+
+        message = Message(self, data)
+        message.channel = self.channels.get(  # type: ignore
+            message.channel_id
+        ) or await self.channels.fetch(
+            message.channel_id
+        )  # type: ignore
```

Comment:
In the "right" scenario, execution might be paused by the `await` so it is quite odd to use a `or` here. I believe it would be clearer to use a explicit `if/else` here.

Reply:
Alright, will do